File: Modules/ServerTypes/Ticket/LogColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LogRow } from "./LogRow";

export interface LogColumns {
    Id: Column<LogRow>;
    StatusName: Column<LogRow>;
    ActionName: Column<LogRow>;
    TicketTitle: Column<LogRow>;
    Username: Column<LogRow>;
    DateCreated: Column<LogRow>;
}

export class LogColumns extends ColumnsBase<LogRow> {
    static readonly columnsKey = 'Ticket.Log';
    static readonly Fields = fieldsProxy<LogColumns>();
}

File: Modules/Ticket/Log/LogColumns.cs
namespace Ticketing.Ticket.Columns;

[ColumnsScript("Ticket.Log")]
[BasedOnRow(typeof(LogRow), CheckNames = true)]
public class LogColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int TicketId { get; set; }    
    public string ActionName { get; set; }
    public string StatusName { get; set; }
    public string Username { get; set; }
    public DateTime DateCreated { get; set; }
}